httpclient/httpclient.go

```diff
@@ -0,0 +1,80 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/satori/go.uuid"
+)
+
+type ClientOpt func(c *http.Client)
+
+func WithTimeout(d time.Duration) ClientOpt {
+	return func(c *http.Client) {
+		c.Timeout = d
+	}
+}
+
+func WithTLSConfig(config *tls.Config) ClientOpt {
+	return func(c *http.Client) {
+		parent := &http.Transport{
+			TLSClientConfig: config,
+		}
+		c.Transport = reqidTransport{parent: parent}
+	}
+}
+
+func WithAuthentication(username, password string) ClientOpt {
+	return func(c *http.Client) {
+		c.Transport = authTransport{
```

Comment:
Don't you override this Transport if you want the client to use both `WithTLSConfig` and `WithAuthentication`?

Reply:
Updated